cg/services/analysis_starter/configurator/implementations/nextflow.py

```diff
@@ -0,0 +1,94 @@
+from pathlib import Path
+
+from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.apps.lims import LimsAPI
+from cg.constants import Workflow
+from cg.constants.priority import SlurmQos
+from cg.models.cg_config import CommonAppConfig
+from cg.services.analysis_starter.configurator.abstract_service import Configurator
+from cg.services.analysis_starter.configurator.extensions.abstract import PipelineExtension
+from cg.services.analysis_starter.configurator.file_creators.config_file import (
+    NextflowConfigFileCreator,
+)
+from cg.services.analysis_starter.configurator.file_creators.params_file.abstract import (
+    ParamsFileCreator,
+)
+from cg.services.analysis_starter.configurator.file_creators.sample_sheet.abstract import (
+    NextflowSampleSheetCreator,
+)
+from cg.services.analysis_starter.configurator.models.nextflow import NextflowCaseConfig
+from cg.store.models import Case
+from cg.store.store import Store
+
+
+class NextflowConfigurator(Configurator):
+
+    def __init__(
+        self,
+        config: CommonAppConfig,
+        store: Store,
+        housekeeper_api: HousekeeperAPI,
+        lims: LimsAPI,
+        config_file_creator: NextflowConfigFileCreator,
+        sample_sheet_creator: NextflowSampleSheetCreator,
+        params_file_creator: ParamsFileCreator,
+        pipeline_extension: PipelineExtension = PipelineExtension(),
+    ):
+        self.root_dir: str = config.root
+        self.store: Store = store
+        self.housekeeper_api: HousekeeperAPI = housekeeper_api
+        self.lims: LimsAPI = lims
+        self.config_file_creator = config_file_creator
+        self.pipeline_extension = pipeline_extension
+        self.sample_sheet_creator = sample_sheet_creator
+        self.params_file_creator = params_file_creator
+
+    def create_config(self, case_id: str) -> NextflowCaseConfig:
+        """Create a Nextflow case config."""
```

Comment:
improve docstring with a numbered list of the steps involved in the configuration. It is not just "creating a config", it involves much more